dashboard/modules/serve/serve_agent.py

```diff
@@ -228,9 +249,15 @@ def submit_config(self, config):
                 ),
             )
 
-        client.deploy_apps(config)
-
-        return Response()
+        try:
+            client.deploy_apps(config)
+        except RayTaskError as e:
```

Comment:
When would this raise a `RayTaskError`? Since `client.deploy_apps()` is nonblocking, is it when the controller is unavailable?

Reply:
In `controller.deploy_apps`, if there is a mode conflict, it will [raise an exception](https://github.com/ray-project/ray/pull/33340/files#diff-e7a727856a2a77fdd45eedf6f91959b303aa9bf99c2fd6b6955d7f7c98e1f3caR497-R507). `client.deploy_apps()` will raise a `RayTaskError` when it tries to wait on `ray.get(controller.deploy_apps())`.

Reply:
Makes sense. Could you add a `Raises` section to the `deploy_apps()` docstring? ([Example](https://github.com/ray-project/ray/blob/1e42e6cd15e2fb96c217cba8484e59ed0ef4b0c8/dashboard/modules/job/sdk.py#L176-L178))

Reply:
Thanks for the suggestion, added!